src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -61,13 +63,18 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                         AuthenticationRequestParameters.RequestContext.CorrelationId,
                         TokenSource.Cache);
                 }
-                else
+                else if(msalAccessTokenItem == null)
```

Comment:
Don't need the `else` clause because the previous clause returns, i.e. this can be just 

`if (msalAccessTokenItem != null)` ... 

Or simpler: `cacheRefresh = (at == null) ? CacheRefresh.NoCacheAt  : CacheRefresh.RefreshIn`